src/test/scala/karazin/scala/users/group/week2/homework/HomeworkSpecification.scala

```diff
@@ -50,28 +50,55 @@ object HomeworkSpecification extends Properties("Homework"):
   }
 
   property("negation") = forAll { (rational: Rational) =>
-    ???
+    val negated = -rational
+    negated.numer == -rational.numer && negated.denom == rational.denom
+
   }
 
   property("addition") = forAll { (left: Rational, right: Rational) =>
-    ???
+    val result = left + right
+    val expectedNumer = left.numer * right.denom + right.numer * left.denom
+    val expectedDenom = left.denom * right.denom
+    val (simplifiedNumer, simplifiedDenom) = simplifyFraction(expectedNumer, expectedDenom)
+    result.numer == simplifiedNumer && result.denom == simplifiedDenom
+
   }
 
   property("subtraction") = forAll { (left: Rational, right: Rational) =>
-    ???
+    val result = left - right
+    val expectedNumer = left.numer * right.denom - right.numer * left.denom
+    val expectedDenom = left.denom * right.denom
+    val (simplifiedNumer, simplifiedDenom) = simplifyFraction(expectedNumer, expectedDenom)
+    result.numer == simplifiedNumer && result.denom == simplifiedDenom
+
   }
 
   property("multiplication") = forAll { (left: Rational, right: Rational) =>
-    ???
+    val result = left * right
+    val expectedNumer = left.numer * right.numer
+    val expectedDenom = left.denom * right.denom
+    val (simplifiedNumer, simplifiedDenom) = simplifyFraction(expectedNumer, expectedDenom)
+    result.numer == simplifiedNumer && result.denom == simplifiedDenom
+
   }
 
   property("division") = forAll { (left: Rational, numer: Int, denom: Int) =>
     val right = Rational(if numer == 0 then 1 else numer, abs(denom) + 1)
-    ???
+    val right = Rational(if numer == 0 then 1 else numer, abs(denom) + 1)
+    val result = left / right
+    val expectedNumer = left.numer * right.denom
+    val expectedDenom = left.denom * right.numer
+    val (simplifiedNumer, simplifiedDenom) = simplifyFraction(expectedNumer, expectedDenom)
+    result.numer == simplifiedNumer && result.denom == simplifiedDenom
+
   }
 
   property("division by zero") = forAll { (left: Rational, int: Int) =>
-    ???
+    throws(classOf[ArithmeticException]) {
+      val right = Rational(1, 0)
```

Comment:
The test is falsified since `Rational(1, 0)` fails with `ArithmeticException`.
`Rational(1, 0)` is not equal to `0`.

Please use `PositiveInteger` for this denom instead.